vision/grid_map/grid_map.py

```diff
@@ -0,0 +1,271 @@
+"""Contains methods to map video with lines from a mask"""
+
+from vision.images import *
+import cv2
+import numpy as np
+
+def image_stitching():
+    """Stitches images together
+    No arguments"""
+    print("Starting image stitching")
+
+    frames = get_all_images("transect", "frames")
+
+    stitcher = cv2.Stitcher.create(mode=cv2.STITCHER_SCANS)
+    ret, stitched_img = stitcher.stitch(frames)
+
+    if ret == cv2.STITCHER_OK:
+        stitched_img = cv2.resize(stitched_img, (0, 0), None, .5, .5)
+        cv2.imshow("Final Image", stitched_img)
+
+        cv2.imwrite(get_folder("transect") + "/stitched_img.jpg", stitched_img)
+
+        cv2.waitKey(0)
+        cv2.destroyAllWindows()
+
+    else:
+        print("Error during stitching")
+
+def draw_lines(frame, mask):
+    """Draws HoughLines on image
+    For example: 'draw_lines(frame, edges)'"""
+    lines = cv2.HoughLinesP(mask, 1, np.pi/180, 100, minLineLength=150, maxLineGap=100)
+
+    if lines is not None:
+        for i in range(len(lines)):
+            line = lines[i]
+
+            x1, y1, x2, y2 = line.reshape(4)
+            cv2.line(frame, (x1, y1), (x2, y2), (0, 255, 0), 2)
+
+class Square:
+    def __init__(self, id, x, y, w, h):
+        self.id = id  # Unique ID for each square
+        self.x = x  # x value
+        self.y = y  # y value
+        self.w = w  # width
+        self.h = h  # height
+        self.visible = True  # True if square is visible in video
+        self.delete = 0  # Tracks num of frames visible = False
+        self.screenshot_num = 0  # Tracks num of screenshots taken per square
+
+squares = []  # Tracks square objects
+id = 0  # Unique ID for each square
+
+def get_contours(mask, frame):
+    curr_squares = []  # Squares in the current frame
+    contours, _ = cv2.findContours(mask, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+
+    for cnt in contours:
+        area = cv2.contourArea(cnt)
+        approx = cv2.approxPolyDP(cnt, .01 * cv2.arcLength(cnt, True), True)
+
+        if area > 10000:
+            # cv2.drawContours(frame, [approx], 0, (255, 0, 0), 5)
+
+            # Check if the contour has 4 sides
+            if len(approx) == 4:
+                x, y, w, h = cv2.boundingRect(approx)
+                aspect_ratio = float(w)/h
+
+                # Check if sides are equal-ish lengths
+                if .5 <= aspect_ratio <= 1.5:
+                    curr_squares.append(Square(-1, x, y, w, h))
+
+    find_squares(curr_squares, frame)
+
+def find_squares(curr_squares, frame):
+    global id
+    global squares  # Square objects already being tracked
```

Comment:
Try to avoid using global variables. Make a class to keep track of these instead.